library/core/src/macros/panic.md

```diff
@@ -9,11 +9,14 @@ tests. `panic!` is closely tied with the `unwrap` method of both
 [`Option`][ounwrap] and [`Result`][runwrap] enums. Both implementations call
 `panic!` when they are set to [`None`] or [`Err`] variants.
 
-This macro is used to inject panic into a Rust thread, causing the thread to
-panic entirely. This macro panics with a string and uses the [`format!`] syntax
-for building the message.
-
-Each thread's panic can be reaped as the [`Box`]`<`[`Any`]`>` type,
+When using `panic!()` you can specify a string payload, that is built using
+the [`format!`] syntax. That payload is used when injecting the panic into
+the calling Rust thread, causing the thread to panic entirely.
+
+The default `std` panic handling strategy is to print the message payload to
+the `stderr` along with the file/line/column information of the `panic!()`
+call. You can override the panic hook using [`std::panic::set_hook()`].
+Inside the hook a panic can be reaped as the [`Box`]`<`[`Any`]`>` type,
```

Comment:
You don't get access to the `Box<Any>` in the hook. You can only look at the payload through [`PanicInfo::payload()`](https://doc.rust-lang.org/stable/std/panic/struct.PanicInfo.html#method.payload), which gives a `&(dyn Any + Send)`. Only when catching a panic (with `catch_unwind` or `join`ing a thread) you get the Box.

Reply:
Oh, I was mislead by the existing documentation. The proposed change does not mention the `Box<Any>` anymore, as this does not integrate well into the rest of the paragraph and probably should be in the docs of `panic::catch_unwind()`].